transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -584,6 +584,17 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
         queryResult
     }
     
+    @Override List<QueryResult> findAllPatientSets(User user) {
+        List resultsWithUserAccess = []
+        List<QueryResult> queryResultList = QtQueryResultInstance.findAll()
```

Comment:
It would be preferable if we can retrieve only the right query instances, instead of retrieving all of them and filtering. For non-admin users that would be retrieving the QtQueryResultInstance's by their queryInstance.userId. That does bypass the access control layer which is unfortunate, but I think it is preferrable to retrieving the potentially large set of query results.